backend/addresses/serializers.py

```
@@ -0,0 +1,81 @@
+        if value.strip().lower() != "egypt":
+            raise serializers.ValidationError("Country must be 'Egypt'.")
+        return value
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        is_default = validated_data.pop("is_default", False)
+
+        address = Address.objects.create(user=user, **validated_data)
+
+        if is_default:
+            address.is_default = True
+            address.save(update_fields=["is_default"])
+
+            try:
+                profile = user.customerprofile
+                profile.default_shipping_address = address
+                profile.save(update_fields=["default_shipping_address"])
+            except CustomerProfile.DoesNotExist:
+                CustomerProfile.objects.create(user=user, default_shipping_address=address)
+
+        return address
+
+    def update(self, instance, validated_data):
+        is_being_defaulted = validated_data.get("is_default", instance.is_default)
+        updated_address = super().update(instance, validated_data)
+
+        if is_being_defaulted:
+            updated_address.is_default = True
+            updated_address.save(update_fields=["is_default"])
+
+            try:
+                profile = updated_address.user.customerprofile
+                profile.default_shipping_address = updated_address
+                profile.save(update_fields=["default_shipping_address"])
+            except CustomerProfile.DoesNotExist:
+                CustomerProfile.objects.create(user=updated_address.user, default_shipping_address=updated_address)
+        else:
+            try:
+                profile = updated_address.user.customerprofile
+                if profile.default_shipping_address == updated_address:
+                    profile.default_shipping_address = None
+                    profile.save(update_fields=["default_shipping_address"])
+            except CustomerProfile.DoesNotExist:
+                pass
```

Again, we need to verify that the user's role is `"customer"`, where we could remove the default address if the
`is_default` value was `False`.

Here are my suggestions:
```suggestion
            if updated_address.user.role == "customer":
                try:
                    profile = updated_address.user.customerprofile
                    if profile.default_shipping_address == updated_address:
                        profile.default_shipping_address = None
                        profile.save(update_fields=["default_shipping_address"])
                except CustomerProfile.DoesNotExist:
                    raise ValidationError("Customer profile must exist to remove default address.")
```

---

Kindly @mohamed952741, this section shall not fail silently, please find my suggestion in the new review.